monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,481 @@
+
+        Outputs: 
+        A JSON datastructure as a file that can be submitted as a part of the test harness to a USS that implements the automatic remote id testing interface. 
+
+        """
+
+        self.flight_telemetry = flight_telemetry
+        self.country_code = country_code
+        self.flight_telemetry_check()
+
+        self.output_directory = Path('test_definitions', self.country_code)
+        # Create test_definition directory if it does not exist
+        self.output_directory.mkdir(parents=True, exist_ok=True)
+        self.output_subdirectories = (Path(self.output_directory, 'aircraft_states'),)
+        for output_subdirectory in self.output_subdirectories:
+            output_subdirectory.mkdir(parents=True, exist_ok=True)
+
+    def flight_telemetry_check(self) -> None:
+        ''' Check if atleast one track is provided, if no tracks are provided, then RIDAircraftState and Test JSON cannot be generated.'''
+
+        # Empty flight points cannot be converted to a Aircraft State, check if the list has
+        if (self.flight_telemetry['telemetery_data_list']):
+            return
+        else:
+            raise ValueError("At least one flight track is necessary to create a AircraftState and a test JSON, please generate the tracks first using AdjacentCircularFlightsSimulator class")
+
+    def write_rid_state(self):
+        ''' This method iterates over flight tracks and geneates AircraftState JSON objects and writes to disk in the test_definitions folder, these files can be used to submit the data in the test harness '''
+
+        reference_time = self.flight_telemetry['reference_time']
+
+        for flight_id, single_flight_telemetry_data in enumerate(self.flight_telemetry['telemetery_data_list']):
+
+            rid_test_file_name = 'flight_' + str(flight_id + 1) + '_rid_aircraft_state' + '.json' # Add 1 to avoid zero based numbering            
+            
+            rid_test_file_path = self.output_subdirectories[0] / rid_test_file_name
+            flight_telemetry_data = {'reference_time': reference_time, 'flight_telemetry': single_flight_telemetry_data}
+            with open(rid_test_file_path, 'w') as f:
+                f.write(json.dumps(flight_telemetry_data))
+
+
+if __name__ == '__main__':
+    # TODO: accept these parameters as values so that other locations can be supplied
+    my_path_generator = AdjacentCircularFlightsSimulator(minx=7.4735784530639648, miny=46.9746744128218410, maxx=7.4786210060119620, maxy=46.9776318195799121, utm_zone='32T')    
+    altitude_of_ground_level_wgs_84 = 570 # height of the geoid above the WGS84 ellipsoid (using EGM 96) for Bern, rom https://geographiclib.sourceforge.io/cgi-bin/GeoidEval?input=46%B056%26%238242%3B53%26%238243%3BN+7%B026%26%238242%3B51%26%238243%3BE&option=Submit
+    COUNTRY_CODE = 'che'
+    flight_points = []
+    query_bboxes = []
```

Comment:
nit: No need to define either `flight_points` or `query_bboxes` here -- `flight_points` is unused, and `query_bboxes` is redefined without reference to this value.

Reply:
Removing, no problems.